src/lua/LuaObject.cpp

```diff
@@ -706,6 +732,96 @@ void LuaObjectBase::RegisterSerializer(const char *type, SerializerPair pair)
 	LUA_DEBUG_END(l, 0);
 }
 
+bool LuaObjectBase::SerializeComponents(LuaWrappable *object, Json &out)
+{
+	lua_State *l = Lua::manager->GetLuaState();
+	LUA_DEBUG_START(l);
+
+	luaL_getsubtable(l, LUA_REGISTRYINDEX, "LuaObjectRegistry");
+
+	// Get the LuaObject for the given LuaWrappable passed in
+	lua_pushlightuserdata(l, object);
+	lua_rawget(l, -2); // Registry, LuaObject
+
+	if (lua_isnil(l, -1)) {
```

Comment:
I wonder if it can happen that the object does not have registration? Is this allowed?

Reply:
It is *technically* impossible for a Body that has been pushed to lua to not be present in the registry, as the only time a Body-derived LuaObject is removed from the registry is after the underlying Body has been deleted and the LuaObject is orphaned.

However, we call SerializeComponents on all Body objects during the save step, so we need to handle the case in which we're serializing a body that didn't get pushed to Lua during its lifetime.